api/envoy/extensions/filters/network/thrift_proxy/v3/thrift_proxy.proto

```diff
@@ -117,6 +117,12 @@ message ThriftProxy {
   // Configuration for :ref:`access logs <arch_overview_access_logs>`
   // emitted by Thrift proxy.
   repeated config.accesslog.v3.AccessLog access_log = 9;
+
+  // If set to true, Envoy will treat header keys in Thrift as case-sensitive instead of converting
+  // them to lower case as per the default behavior.
```

Comment:
Is this comment correct? 

Internally (e.g. for route machine, substation formatters, etc) Envoy will still use case-insensitive header keys stripped of NUL, CR and LF. My understanding is that enabling this flag will cause forwarded header keys to retain their original formatting.

Reply:
Vaguely correct, if anything. How about we go with something like the following (reusing most of your words):

If set to true Envoy will forward headers retaining their original formatting while internally route matching and other filters will treat them as case-insensitive and they'll also be stripped of NUL, CR and LF.

cc: @wbpcode @fishy @fishcakez

Reply:
should we rename this to `header_keys_preserve_case` and then comment like `If set to true, Envoy will preserve the case of Thrift header keys instead of serializing them to lower case as per the default behavior`.

The header keys aren't actually case sensitive internally because we're using the http matchers etc?

Reply:
Agree with both suggestions.

Reply:
Cool, thanks all. Updated.